src/Dialogs/StripeDialog.vala

```diff
@@ -427,5 +430,42 @@ public class AppCenter.Widgets.StripeDialog : Gtk.Dialog {
 
         return data.str;
     }
+
+    private string get_stripe_error (Json.Object error_object) {
+        unowned string error_type = error_object.get_string_member ("type");
+        debug ("Stripe error type: %s", error_type);
+        switch (error_type) {
+            case "card_error":
+                unowned string error_code = error_object.get_string_member ("code");
+                debug ("Stripe error code: %s", error_code);
+                switch (error_code) {
+                    case "incorrect_number":
+                    case "invalid_number":
+                        return _("Please review your payment info number and try again.");
+                    case "invalid_expiry_month":
+                    case "invalid_expiry_year":
+                        return _("Please review your payment info expiry and try again.");
+                    case "incorrect_cvc":
+                    case "invalid_cvc":
+                        return _("Please review your payment info CVC and try again.");
+                    case "expired_card":
+                        return _("The provided payment info has expired.");
+                    case "processing_error":
+                        return _("An payment processing error occurred, please retry later. We apologize for any inconvenience caused.");
```

Comment:
"An error occurred while processing the card. Please try again later. We apologize for any inconvenience caused."